### PR TITLE
Fix tcp rendezvous disconnects issue

### PR DESCRIPTION
**Pull request by canndrew**
This change is [Reviewable](https://reviewable.io/reviews/maidsafe/nat_traversal/60)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @vinipsmaker

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by canndrew** [comment]
CI tests are failing for windows :(
I'll have to get my windows VM working again to debug this.

**Comment by vinipsmaker** [comment]
> I'll have to get my windows VM working again to debug this.

Let me know when the PR is ready for review.

File: src/mapped_tcp_socket.rs [hunk 1]
```diff
@@ -783,33 +792,112 @@ pub fn tcp_punch_hole(socket: net2::TcpBuilder,
...
+                                if this_sum > top_sum {
+                                    Some((this_stream, this_sum))
+                                }
+                                else {
+                                    Some((top_stream, top_sum))
+                                }
+                            },
+                            None => Some((this_stream, this_sum))
+                        }
+                    });
```
Comment (vinipsmaker) [inline src/mapped_tcp_socket.rs:896]: Care to elaborate what you're doing here? Are you dropping all but the last responsive stream? A comment explaining a little of context why this is needed would help a lot.

Comment (canndrew) [inline src/mapped_tcp_socket.rs:896]: Good point. I've added some comments.

**Comment by vinipsmaker** [comment]
Reviewed 3 of 4 files at r1, 1 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/nat_traversal/60)_

**Pull request merged**